pkg/webhook/kubernetes/tls.go

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright (c) 2012-2019 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package webhook_k8s
+
+import (
+	"context"
+
+	"github.com/devfile/devworkspace-operator/pkg/webhook/service"
+
+	"github.com/devfile/devworkspace-operator/pkg/kubernetes/tls"
+	"github.com/devfile/devworkspace-operator/webhook/server"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = logf.Log.WithName("webhook-k8s")
+
+// SetupSecureService handles TLS secrets required for deployment on Kubernetes.
+func SetupSecureService(client crclient.Client, ctx context.Context, namespace string) error {
+	devworkspaceSecret := &corev1.Secret{}
+
+	err := client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: tls.TLSSelfSignedCertificateSecretName}, devworkspaceSecret)
+	if err == nil {
+		return nil
```

Comment:
Should we also check that service is in place to recover after unproper uninstallation?

Reply:
Not sure I understand what you mean, if the service exists then it's just going to be updated. Or did you mean something else?

Reply:
>  if the service exists then it's just going to be updated

Isn't it supposed to be updated below here https://github.com/devfile/devworkspace-operator/pull/159/files/d4e58c25e4d83adc706cfaaac3f8ec4587e015bb#diff-3dee34c4acc1e59056a87b123e78432aR46? 

If yes, then how it's going to be updated if we return here?

Reply:
Yes, I understand that. I just don't understand what you mean by
> Should we also check that service is in place to recover after unproper uninstallation

As in check that the service exists and if it does delete before erroring? 

If it errors here there then the webhook server deployment just won't get setup

Reply:
I mean that there can be cases when TLSSelfSignedCertificateSecretName exists but service itself is removed.
and now it seems to be handled properly apart from fact that certs are going to be generated each time
So, code proposal instead of thousand words
https://github.com/devfile/devworkspace-operator/compare/kubernetes-cert-job...sleshchenko:kubernetes-cert-job-proposal2?expand=1

Reply:
Ah okay that makes sense. Yeah I'll add that in